Tests/AppStoreConnectClientTests/RateLimitMiddlewareTests.swift

```diff
@@ -0,0 +1,166 @@
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Affero General Public License for more details.
+//
+//  You should have received a copy of the GNU Affero General Public License
+//  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+//
+//  RateLimitMiddlewareTests.swift
+//
+//
+//  Created by Mykola Vasyk on 15.05.2024.
+//
+
+import XCTest
+import HTTPTypes
+import OpenAPIRuntime
+@testable import AppStoreConnectClient
+
+final class RateLimitMiddlewareTests: XCTestCase {
+    func testRateLimitAllowsRequests() async throws {
+        let request = HTTPTypes.HTTPRequest(
+            method: .get,
+            scheme: "http",
+            authority: "example.com",
+            path: "/test"
+        )
+        var fields = HTTPFields()
+        let unwrap = try XCTUnwrap(HTTPField.Name("x-rate-limit"))
+        fields[unwrap] = "user-hour-lim:1111; user-hour-rem:222"
+        let baseURL = try XCTUnwrap(URL(string: "http://example.com"), "Unwrap fail")
+        let httpResponse = HTTPTypes.HTTPResponse(
+            status: .accepted,
+            headerFields: HTTPFields(fields[fields: unwrap])
+        )
+        let sut = RateLimitMiddleware()
+        do {
+            let result = try await sut.intercept(
+                request,
+                body: nil,
+                baseURL: baseURL,
+                operationID: "someOperationID"
+            ) { _, _, _ in
+                return (httpResponse, nil)
+            }
+            XCTAssertEqual(result.0.status, .accepted)
+        } catch {
+            XCTFail("Unexpected error: \(error)")
+        }
+    }
+    
+    func testRateLimitExceededError() async throws {
+        let expectedLimit = 1111
+        let expectedRemaining = 0
+        let request = HTTPTypes.HTTPRequest(
+            method: .get,
+            scheme: "http",
+            authority: "example.com",
+            path: "/test"
+        )
+        var fields = HTTPFields()
+        let unwrap = try XCTUnwrap(HTTPField.Name("x-rate-limit"))
+        fields[unwrap] = "user-hour-lim:\(expectedLimit); user-hour-rem:\(expectedRemaining)"
+        let baseURL = try XCTUnwrap(URL(string: "http://example.com"), "Unwrap fail")
+        let httpResponse = HTTPTypes.HTTPResponse(
+            status: .accepted,
+            headerFields: HTTPFields(fields[fields: unwrap])
+        )
+        let sut = RateLimitMiddleware()
+        do {
+            let result = try await sut.intercept(
+                request,
+                body: nil,
+                baseURL: baseURL,
+                operationID: "someOperationID"
+            ) { _, _, _ in
+                return (httpResponse, nil)
+            }
+            XCTFail("Expected error not thrown, nevertheless got result: \(result)")
+        } catch RateLimitError.rateLimitExceeded(let remaining, let limit) {
```

Comment:
Need implementation like other tests 
Something like this
} catch {
            XCTAssertEqual(error as? RateLimitError, .rateLimitExceeded(remaining, limit))
        }

Reply:
I implemented it like this:
} catch {
            if case let RateLimitError.rateLimitExceeded(remaining, limit) = error {
                XCTAssertEqual(error as? RateLimitError, .rateLimitExceeded(remaining: remaining, from: limit))
                XCTAssertEqual(remaining, expectedRemaining)
                XCTAssertEqual(limit, expectedLimit)
            } else {
                XCTFail("Unexpected error: \(error)")
            }